runtime/kubernetes/namespace.go

```diff
@@ -0,0 +1,116 @@
+package kubernetes
+
+import (
+	"strings"
+
+	"github.com/micro/go-micro/v3/logger"
+	"github.com/micro/go-micro/v3/util/kubernetes/client"
+)
+
+func (k *kubernetes) ensureNamepaceExists(ns string) error {
```

Comment:
I'm not against the concept of namespaces being in go micro as I think it's a perfectly valid thing to have in OSS, but I thought we try to hide it in general?

Reply:
This function already existed, I just moved all the namespace related cruft to a namespace.go file. Chris' PR will tidy this up further.

Reply:
This function is to ensure the k8s namespace exists before the runtime creates a service & deployment within it